src/core/unit.c

```diff
@@ -327,6 +327,9 @@ int unit_set_description(Unit *u, const char *description) {
 
         assert(u);
 
+        if (streq_ptr(u->description, description))
+                return 0;
+
         r = free_and_strdup(&u->description, empty_to_null(description));
```

Comment:
Does this make any difference? free_and_strdup() already does nothing and returns 0 if the strings are the same.

Reply:
I've dropped the patch and tested again and it seems that it is indeed not needed. Good catch. Thanks!